### PR TITLE
Fix warnings and errors for ceres & eigen local submodule.

### PR DESCRIPTION
**Pull request by cbalint13**
Folks,

 This PR should fix two minor issues:

  **1. Ceres submodule is telling us:**

-----<<<--------
CMake Warning (dev) at third_party/ceres-solver/CMakeLists.txt:39 (PROJECT):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:
    CERES_VERSION
This warning is for project developers.  Use -Wno-dev to suppress it.
----->>>----------

  **2. Eigen submodule is telling us:**
-----<<<--------
abort: repository /home/cbalint/work/openMVG/src not found!
abort: repository /home/cbalint/work/openMVG/src not found!
----->>>----------

I am using standard ways (no extra args to cmake):
cmake-3.2.2-1.fc22.x86_64
/build# cmake ../src

*MOTIVATION:*

* As release engineering point of view i am trying to pack opeMVG for RedHat/Fedora products, but seems openMVG is not quite ready for such task. If agree, more fixes like this one will come in small subsequent PRs. Will try small but clear PR's one at a time to be easy to track. First I try eliminate warns like these ones then will try teach openMVG build suite to be able take all libraries from external paths too (a **must** requirement for packaging). Intentions are to add optional features by using LIBXYZ_HINT_DIRS (actually ceres, eigen3 can be specified ext path right now).

### CODE REVIEW

**Comment by cbalint13** [comment]
* Also, last goal would be openMVG shared library.

**Comment by fabiencastan** [comment]
Allow to build openMVG in shared would be really great!

**Comment by cbalint13** [comment]
@fabiencastan ,

  Patches are already done in a series of more upcoming PRs. Will do more PR to allow better review and feedback time between changes. Shared library support is also **must** condition for packaging to any modern distro.

File: src/third_party/eigen/CMakeLists.txt [hunk 1]
```diff
@@ -36,11 +36,6 @@ string(REGEX MATCH "define[ \t]+EIGEN_MINOR_VERSION[ \t]+([0-9]+)" _eigen_minor_
 set(EIGEN_MINOR_VERSION "${CMAKE_MATCH_1}")
 set(EIGEN_VERSION_NUMBER ${EIGEN_WORLD_VERSION}.${EIGEN_MAJOR_VERSION}.${EIGEN_MINOR_VERSION})
 
-# if the mercurial program is absent, this will leave the EIGEN_HG_CHANGESET string empty,
-# but won't stop CMake.
-execute_process(COMMAND hg tip -R ${CMAKE_SOURCE_DIR} OUTPUT_VARIABLE EIGEN_HGTIP_OUTPUT)
```
Comment (pmoulon) [inline src/third_party/eigen/CMakeLists.txt]: I think the ${CMAKE_SOURCE_DIR} should be replaced by ${CMAKE_CURRENT_SOURCE_DIR }

**Comment by cbalint13** [comment]
@pmoulon ,

 Those lines containing mentioned CMAKE_SOURCE_DIR are marked in the patch to be removed. They create noise and make cmake > 3.0.0 to stop/fail, their behaviour is unexpected across various platforms. I also believe that those lines are superfluous and nonsense for booth internal and external eigen cases.

**Comment by cbalint13** [comment]
@pmoulon ,

* Squashed all into 4d93fd8 for better view.

**Comment by pmoulon** [comment]
Merged!
Happy to see you back on OpenMVG @cbalint13

**Pull request merged**